auction-app-frontend/src/components/InputField/InputField.jsx

```diff
@@ -1,24 +1,52 @@
 import { useFormContext } from "react-hook-form";
+import { useState } from "react";
 
-import { SelectField } from "src/components"; 
+import { SelectField, CustomFileUploader } from "src/components"; 
 
 import "./style.scss";
 
-const InputField = ({ name, label, type, rules, step, options, className }) => {
-    const { register, formState: { errors } } = useFormContext();
+const InputField = ({ name, label, type, rules, step, options, className, placeholder, onSelectChange }) => {
+    const { register, setValue, formState: { errors } } = useFormContext();
+
+    const [dateValue, setDateValue] = useState(""); 
+    const [files, setFiles] = useState([]);
+        
+    const onChangeDate = (event) => {
+        setDateValue(event.target.value);
+        setValue(name, event.target.value);
+    };
 
     const renderField = () => {
         const inputProps = {
             ...register(name, { ...rules }),
             id: name,
             type: type,
             className: errors[name] ? "error" : "",
-            ...(step && { step }) 
+            ...(step && { step }) ,
+            placeholder: placeholder || ""
         };
     
         switch (type) {
+            case "date": 
```

Comment:
Why do we have it like this, and not like separate components, I mean, Textarea and Input field aren't the same, also dropdown or file upload component. 
This component isn't anymore InputField, it doesn't have the same purpose.